packages/langium/src/grammar/grammar-util.ts

```diff
@@ -322,6 +322,25 @@ export function getTypeName(type: ast.AbstractType | ast.InferredType): string {
     throw new TypeResolutionError('Cannot get name of Unknown Type', type.$cstNode);
 }
 
+export function getExplicitRuleTypeName(rule: ast.ParserRule): string | undefined {
+    if (rule.inferredType) {
+        return rule.inferredType.name;
+    } else if (rule.dataType) {
+        return rule.name;
```

Comment:
Seems like this is the only change here. If that's the case, we can pull this out like what's done below:
https://github.com/langium/langium/blob/17f062935c2915889cf5a1def5bbddf90fa3fea2/packages/langium/src/grammar/grammar-util.ts#L376
And in such a case, that change could be lifted up to the portion noted above in the previous comment, and this function could be dropped -- especially since it duplicates most of what's already present in `getExplicitRuleType` below.

Reply:
I used your example and fitted the check into the code using a ternary operator. It works and should be less complex than before now.